### ISSUE TITLE
New Machine requirement: Windows server 2022 build systems

### ISSUE THREAD

**Issue by sxa** [adoptium/infrastructure#2938]
I need to request a new machine: As discussed in this week's PMC meeting we intende to move from building on Windows Server 2012 to the latest version now that @andrew-m-leonard has determined that we can produce binary identical builds of the latest JDKs across both versions.

- New machine operating system (e.g. linux/windows/macos/solaris/aix): windows
- New machine architecture (e.g. x64/aarch32/arm32/ppc64/ppc64le/sparc): x64
- Provider (leave blank if it does not matter): Azure
- Desired usage: build (and some test!)
- Any unusual specification/setup required: We'll want to use the VS2019 fixed layout for the install
- How many of them are required: 2 for build, maybe more for test (to replace existing ones)

Please explain what this machine is needed for: Building and testing Windows Server 2022 when 2012 goes out of support later this year.

**Comment by sxa** [comment]
Provisioned build-azure-win2022-x64-1 (`172.187.129.163`)
No setup done yet (Will discuss with @andrew-m-leonard prior to deploying compiler)
Nagios and WinRM ports are enabled

**Comment by sxa** [comment]
Provisioned a 4 core version at `172.187.176.15` (Earlier one was 2 core which would be very low for our typical build machines. WinRM and NSClient (12489) ports are enables.

**Comment by sxa** [comment]
These machines need to be deployed using the VS2019 layout from `/Vendor_Files/windows/VS2019_layout.tar.gz` on the AWX server. [This issue](https://github.com/adoptium/temurin-build/issues/2978) covers what was done to create and deploy that image. We need to distill that into the commands required to copy the layout across from AWX to the windows box and deploy it in place of the existing VS2019 role (ideally falling back to the community edition install/download if we don't have a layout file so that others can still use the playbooks)
This will enable reproducibility of the builds.

**Comment by steelhead31** [comment]
TLDR; - Install from layout into default program files(x86) path , after discussion detailed below.

**Summary Of Slack Discussion About Install Path From Layout.**  
 

All, Im currently implementing some changes to the ansible infrastructure windows playbooks regarding the Visual Studio 2019 install , currently the Visual Studio 2019 is installed into its default location, ie c:\\program files\\visual studio\\2019,  however as part of the windows reproducible build work, the default installation , when installing from a pre-defined layout is c:\\vs2019 ...   the new playbook changes, has the capability to install from a predefined layout, where available, and "fall back" to the download and install ( which has issues for reproducibility )...   are there any thoughts on whether the installation paths should be aligned for both scenarios?, and if so , which way should they be aligned ?,  a preference for c:\\program files OR c:\\vs2019 ?

**leonarda**  [23 hours ago](https://adoptium.slack.com/archives/C09NW3L2J/p1678353214662619?thread_ts=1678352813.374699&cid=C09NW3L2J)  
I always prefer short vs2019, but that’s a personal setup, if one needs to find it the first place to look is Program Files…

**sxa**  [23 hours ago](https://adoptium.slack.com/archives/C09NW3L2J/p1678353483145849?thread_ts=1678352813.374699&cid=C09NW3L2J)  
For the latter of those two reasons, and avoiding any risk of it not being detected properly by any of the processes, my preference would be to keep it in the current directory under `Program Files` - avoids any issues with the detection and any risk of ending up with two conflicting versions on the machine which could make problem determination harder.

---

**New**

![](https://ca.slack-edge.com/T09NLKFT4-U02JZQXAW0L-e783260de8df-48)

**Jie Kang**  [13 hours ago](https://adoptium.slack.com/archives/C09NW3L2J/p1678389792560169?thread_ts=1678352813.374699&cid=C09NW3L2J)  
 

If not done already, I recommend summarizing such discussions and decisions in the relevant issue so that it's more easily seen in the future.

**Comment by sxa** [comment]
Testing (Note that in the absence of the `win2012` build tag this machine will not automatically be used for builds yet. Verifying a few here by labelling them directly targetting the new machine:
- [jdk8u-windows-x64-temurin](https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-windows-x64-temurin/270/console)
- [jdk11u-windows-x64-temurin](https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk11u/job/jdk11u-windows-x64-temurin/210/console)
- [jdk20-windows-x64-temurin](https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk20/job/jdk20-windows-x64-temurin/)
- [jdk8u-windows-x86-32-temurin](https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-windows-x86-32-temurin/259/console)
- 
FYI @andrew-m-leonard These aren't reproducible tests, but just verifying that three of the versions will build ok (Correct compilers available etc.)

**Comment by sxa** [comment]
Running on the `-2` machine:
- https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-windows-x64-temurin/271/console

Failed due to potential permissions issue:
```
13:27:03  [CHECKOUT] Checking out Adopt Pipelines https://github.com/adoptium/ci-jenkins-pipelines.git : master
[Pipeline] checkout
13:27:03  The recommended git tool is: git
13:27:09  No credentials specified
13:27:09  Cloning the remote Git repository
13:27:09  Cloning repository https://github.com/adoptium/ci-jenkins-pipelines.git
13:27:10  ERROR: Error cloning remote repo 'origin'
13:27:10  hudson.plugins.git.GitException: Command "git fetch --tags --force --progress -- https://github.com/adoptium/ci-jenkins-pipelines.git +refs/heads/*:refs/remotes/origin/*" returned status code 128:
13:27:10  stdout: 
13:27:10  stderr: fatal: detected dubious ownership in repository at '/cygdrive/c/workspace/openjdk-build'
13:27:10  To add an exception for this directory, call:
13:27:10  
13:27:10  	git config --global --add safe.directory /cygdrive/c/workspace/openjdk-build
13:27:10  
```

**Comment by sxa** [comment]
Second machine now fixed:
- https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-windows-x64-temurin/283/console
- https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-windows-x86-32-temurin/260/console

**Comment by sxa** [comment]
Both of those are showing as aborted but it looks like that might have been something downstream in the test jobs.

I'll run again: 
- https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-windows-x64-temurin/285/
- https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-windows-x86-32-temurin/

There should be new runs of both of those jobs in tonight's scheduled run so we will be able to directly compare.

**Comment by andrew-m-leonard** [comment]
@sxa Test job Aborted due to this.. Windows docker node ??? :
```
18:54:49  There are no nodes with the label ‘[ci.role.test&&hw.arch.x86&&sw.os.windows&&sw.tool.docker](https://ci.adoptium.net/label/ci.role.test&&hw.arch.x86&&sw.os.windows&&sw.tool.docker/)’
```

**Comment by sxa** [comment]
> @sxa Test job Aborted due to this.. Windows docker node ??? :

Are you talking about the [sanity.external one](https://ci.adoptium.net/job/Test_openjdk8_hs_sanity.external_x86-64_windows/13/console)? I don't think that's generally run on Windows so can be ignored

**Comment by sxa** [comment]
Although [sanity.openjdk](https://ci.adoptium.net/job/Test_openjdk8_hs_sanity.openjdk_x86-64_windows/775/) isn't looking too happy either - some java_util stream failures.

**Comment by sxa** [comment]
https://github.com/adoptium/ci-jenkins-pipelines/pull/659 will allow the JDK20+ jobs to run on these new machines
